src/langchain_chatbot/components/langchain_utils_with_local_llm.py

```diff
@@ -0,0 +1,76 @@
+# type: ignore
+import streamlit as st
+from prompts import final_prompt, answer_prompt
+from table_details import table_chain as select_table
+from vector_store import retriever_prompt
+from langchain_core.runnables import RunnablePassthrough,RunnableMap
+from langchain_core.output_parsers import StrOutputParser
+from operator import itemgetter
+from langchain.memory import ChatMessageHistory
+from langchain_community.tools.sql_database.tool import QuerySQLDataBaseTool
+from langchain_openai import ChatOpenAI
+from langchain.chains import create_sql_query_chain
+from langchain_community.utilities.sql_database import SQLDatabase
+import os
+from vectors_store_sentence_transformer import retriever
+from dotenv import load_dotenv
+load_dotenv()
+
+OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
+LANGCHAIN_TRACING_V2 = os.getenv("LANGCHAIN_TRACING_V2")
+LANGCHAIN_API_KEY = os.getenv("LANGCHAIN_API_KEY")
+db_url = os.getenv("DB_URL")
+
+@st.cache_resource
+def get_chain():
+    print("Creating chain")
+    db = SQLDatabase.from_uri(db_url)
+    llm = ChatOpenAI(model="gpt-3.5-turbo", temperature=0)
+    context_chain = (
+        RunnableMap({"context": itemgetter("context"), "question": itemgetter("question")})
+        | retriever_prompt
```

Comment:
why we need retriever_prompt ? we already have the context.

retriever_prompt used to choose the right tables using llm , but we dont need it anymore :)